### ISSUE TITLE
Merge deliver analysis commands

### ISSUE THREAD

**Issue by moonso** [Clinical-Genomics/cg#806]
## Current situation

ATM there are four more or less identical delivery commands implemented for the workflows:

- https://github.com/Clinical-Genomics/cg/blob/master/cg/cli/workflow/balsamic/deliver.py
- https://github.com/Clinical-Genomics/cg/blob/master/cg/cli/workflow/microsalt/deliver.py
- https://github.com/Clinical-Genomics/cg/blob/master/cg/cli/workflow/mip_dna/deliver.py
- https://github.com/Clinical-Genomics/cg/blob/master/cg/cli/workflow/mip_rna/deliver.py

## Suggested solution

These pieces of code should be merged into one new command called `cg deliver analysis` that take case_id as an argument and bundle as a option. The options could be in 

```python
["balsamic", "balsamic-qc", "microsalt", "microsalt-qc", "mip-dna", "mip-rna"]
```

The new command `cg deliver` will also include `cg deliver fastq`, related to #794

**Comment by moonso** [comment]
# Tags for delivery types

## Balsamic
 
### balsamic

```python
[
    "clinical-vcf-pass", 
    "clinical-vcf-pass-index", 
    "cnv-gene-metrics", 
    "cnv-scatter", 
    "cnv-diagram"
    "multiqc-html", 
    "tumor-bam" OR "tumor-cram", 
    "normal-bam" OR "normal-cram", 
]
```

### balsamic-qc

@hassanfa @emmser @keyvanelhami can you help to fill in here

## microsalt

### microsalt

```python
[
    "microsalt-qc",
    "microsalt-type",
    "assembly",
    "trimmed-forward-reads",
    "trimmed-reverse-reads",
    "reference-alignment-deduplicated"
]
```

### microsalt-qc

@talnor @sylvinite could you help to fill here

**Comment by keyvanelhami** [comment]
@moonso do you want us to add what files to deliver in `balsamic-qc` ? Or what qc parameters to check for if the analysis passed or not?

**Comment by moonso** [comment]
> do you want us to add what files to deliver in balsamic-qc ?

Yes what tags to use when delivering balsamic-qc

**Comment by moonso** [comment]
Ok so @emmser @keyvanelhami @hassanfa  @talnor @sylvinite I just realised that we might need to have another layer of delivery tags. Some files should be delivered per sample and some per case as you can see in the MIP example above. Please clearify if the tags are on sample- or case level. Thx

**Comment by hassanfa** [comment]
I'm not sure I can look into this soon, but maybe next week? Unless it is blocking your progress and implementation.

**Comment by moonso** [comment]
> I'm not sure I can look into this soon, but maybe next week? Unless it is blocking your progress and implementation.

Next week is fine

**Comment by keyvanelhami** [comment]
@moonso I updated what files that should be delivered on sample level and case level. 

What do you mean with `balsamic-qc` ? Do you mean which qc-files that are going to be delivered? The only qc file to be delivered is `multiqc-html` which is already in the list

**Comment by moonso** [comment]
> @moonso I updated what files that should be delivered on sample level and case level.

Great!

> What do you mean with balsamic-qc ? Do you mean which qc-files that are going to be delivered? The only qc file to be delivered is multiqc-html which is already in the list

If you want to have two types of delivery. One that include many files and one that only include a qc-report. In that case you could add only that tag to the balsamic-qc delivery

**Comment by hassanfa** [comment]
So as discussed this morning, Keyvan will go through the items in tags and add more relevant ones:

1. Move sample level tags from case to sample
2. Confirm these tags exist: `quality-trimmed-fastq-read1` and `quality-trimmed-fastq-read2` and add them to sample level

**Comment by keyvanelhami** [comment]
@moonso  Looking at housekeeper, Balsamic analysis doesn't have a combination tag called `clinical-vcf-pass`. Take a look yourself: `housekeeper get bundle gamesloth- v`

Do you require a combination tag or can I add below files individually?
```
Vardict: annotated-somatic-vcf-pass
Manta: annotated-somatic-vcf-pass
CNV: annotated-somatic-vcf-pass
```

**Comment by hassanfa** [comment]
I looked through the tags for BALSAMIC, and it'll take time to prepare a proper single unique tag. The current design does not allow for such change. We have to go ahead with a tuple of tags. And an empty `SAMPLE_TAGS` since some files don't have proper case or sample tag in housekeeper.

### balsamic

```python
CASE_TAGS =[
    ("cnv", "cnvkit", "vcf-all", "annotated-somatic-vcf-all"),
    ("cnv", "cnvkit", "vcf-all", "annotated-somatic-vcf-all-index"),
    ("vardict", "snv", "annotated-somatic-vcf-pass, vcf-pass"),
    ("vardict", "snv", "annotated-somatic-vcf-pass-index", "vcf-pass"),
    ("manta", "sv", "annotated-somatic-vcf-pass", "vcf-pass"),
    ("manta", "sv", "annotated-somatic-vcf-pass-index", "vcf-pass"),
    ("cnv-scatter",),
    ("cnv-diagram",),
    ("multiqc-html",),
    ("balsamic-report",),
    ("normal-cram",), #if normal sample exists
    ("normal-cram-index",), #if normal sample exists
    ("tumor-cram",),
    ("tumor-cram-index",),
]
SAMPLE_TAGS = []
```

### balsamic-qc for Panel and WES

```python
CASE_TAGS =[
    ("multiqc-html",),
]
SAMPLE_TAGS = [
("quality-trimmed-fastq-read1",), #also do the same if a normal sample exists
("quality-trimmed-fastq-read2",), #also do the same if a normal sample exists
]
```

### balsamic-qc for WGS

```python
CASE_TAGS =[
    ("multiqc-html",),
]

SAMPLE_TAGS = [
...
# concatenated or non-concatenated fastq files same as the ones we deliver to BALSAMIC
]
```

@moonso Does this work for your PR?

**Comment by moonso** [comment]
Closed by #812
